src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,161 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
+                    {
+                        // Validation to ensure that entities have valid columns matching their source fields.
+                        if (!sqlMetadataProvider.TryGetBackingColumn(entityName, sourceField, out _))
+                        {
+                            HandleOrRecordException(new DataApiBuilderException(
+                                message: $"Entity: {entityName} has a relationship: {relationshipName} with source field: {sourceField} that " +
+                                    $"does not exist as a column in {entityName}.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+                }
+
+                // In all kinds of relationships, if targetFields are included they must be valid columns in the backend.
+                if (relationship.TargetFields is not null)
+                {
+                    foreach (string targetField in relationship.TargetFields)
+                    {
+                        if (!sqlMetadataProvider.TryGetBackingColumn(relationship.TargetEntity, targetField, out _))
+                        {
+                            // Validation to ensure that entities that have target fields defined, define target fields
+                            // that are valid columns in the target entity.
+                            HandleOrRecordException(new DataApiBuilderException(
+                                message: $"Entity: {entityName} has a relationship: {relationshipName} with target field: {targetField} that " +
+                                    $"does not exist as a column in target entity: {relationship.TargetEntity}.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+                }
+
+                // Linking object exists and we therefore have a many to many relationship. Validation here differs from one to many and many to one in that
+                // the source and target fields are now only indirectly related through the linking object. Therefore, it is the source and linkingSource
+                // along with the target and linkingTarget fields that must match, respectively. Source and linkingSource fields provide the relationship
+                // from the source entity to the linkingObject while target and linkingTarget fields provide the relationship from the target entity to the
+                // linkingObject.
+                // see: https://learn.microsoft.com/en-us/azure/data-api-builder/relationships#many-to-many-relationship
+                if (relationship.LinkingObject is not null)
```

Comment:
line 916 to line 986, can be shortened. there is many repetative code.
```
if (relationship.LinkingObject is not null)
{
    ValidateFields(relationship.SourceFields, relationship.LinkingSourceFields, "source", entityName, relationshipName);
    ValidateFields(relationship.TargetFields, relationship.LinkingTargetFields, "target", entityName, relationshipName);
}

private void ValidateFields(string[]? fields, string[]? linkingFields, string fieldType, string entityName, string relationshipName)
{
    if (fields is not null && linkingFields is null)
    {
        HandleOrRecordException(CreateException($"{fieldType} fields", "linking " + fieldType + " fields", entityName, relationshipName));
    }

    if (linkingFields is not null && fields is null)
    {
        HandleOrRecordException(CreateException("linking " + fieldType + " fields", $"{fieldType} fields", entityName, relationshipName));
    }

    if (fields is not null && linkingFields is not null && fields.Length != linkingFields.Length)
    {
        HandleOrRecordException(new DataApiBuilderException(
            message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with {fields.Length} " +
                $"{fieldType} fields defined, but {linkingFields.Length} linking {fieldType} fields defined.",
            statusCode: HttpStatusCode.ServiceUnavailable,
            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
    }
}

private DataApiBuilderException CreateException(string nonNullFields, string nullFields, string entityName, string relationshipName)
{
    return new DataApiBuilderException(
        message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with {nonNullFields} that are not null, " +
            $"but {nullFields} that are null.",
        statusCode: HttpStatusCode.ServiceUnavailable,
        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
}
```

Reply:
thanks for these suggestions, Ive added the helper, but made the error message a little different so that we can use XOR in the comparison.